docs(changelog): add 8.4.1 release notes (#5055)
canonical/snapcraft: docs/reference/changelog.rst

  For a complete list of commits, check out the `X.Y.Z`_ release on GitHub.

8.4.1 (2024-Sep-20)
-------------------

Core
====

* Fix a regression where numeric entries in a ``snapcraft.yaml`` could not be
  parsed.

Bases
#####

core24
""""""

* Fix a regression where ``build-for`` could not be omitted in a ``platforms``
  entry in a ``snapcraft.yaml`` file.

* Fix a regression where ``--shell`` and ``--shell-after`` were not supported
  for the ``pack`` command (`#4963`_).

> * Fix a regression where ``--debug`` would not shell into the build environment

Reviewer: ```suggestion
* Fix a regression where ``--debug`` wouldn't open a shell into the build environment
```

  if the packing step fails (`#4959`_).

Plugins
#######

NPM
"""

* Fix a bug where NPM parts fail to build if the ``pull`` and ``build`` steps

>   did not occur in the same execution of Snapcraft.

Reviewer: ```suggestion
  didn't occur in the same execution of Snapcraft.
```

>   did not occur in the same execution of Snapcraft.

Reviewer: > execution

Do we mean "instance"? I'm not a fan of "execution" as a correlate of "executable".

Author: Yeah, I think "instance" is a better choice.

Command line
============

* Fix a regression where store errors would be raised as an internal error
  (`#4930`_).

* Add documentation links for error messages about using an `ESM base`_.

Remote build
============

> * Fix a regression where ``--build-for`` and ``--platform`` could not accept

Reviewer: ```suggestion
* Fix a regression where ``--build-for`` and ``--platform`` couldn't accept
```

  comma-separated values (`#4990`_).

* Fix a regression where remote build errors would be raised as an internal
  error (`#4908`_).

* Add documentation links and recommended resolutions to remote-build errors.

Store
=====

> * Fix a regression where Ubuntu One macaroons could not be refreshed

Reviewer: ```suggestion
* Fix a regression where Ubuntu One macaroons couldn't be refreshed
```

  (`#5048`_).

> For a complete list of commits, check out the `8.4.1`_ release on GitHub.

Reviewer: ```suggestion
For a complete list of changes, check out the `8.4.1`_ release on GitHub.
```

I noticed this for the first time. Maybe a nitpick, but I feel that "changes" is better here because a commit is an implementation detail of our VCS. This will become more relevant next cycle when we refine the release note template.

Author: I like the idea of abstracting the changes from the VCS, but on the other hand your suggestion eludes that there are changes that are not documented in the changelog.
